app/models/task.py

```diff
@@ -1,5 +1,29 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from datetime import datetime
+from typing import Optional
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at : Mapped[datetime | None] = mapped_column(nullable = True, default = None)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def to_dict(self):
+        return {
+            "id" : self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None
+        }
```

Comment:
Nitpick: blank line should surround methods
```suggestion
        }
        
```

Reply:
How could you add logic here so that you can conditionally set the goal attribute on the task instance if you need to? Currently this logic is done in your route (see my comment for lines 72-80 in `goal_routes.py`, but I'd prefer that logic to live here so that your route logic is more concise and so you don't need to repeat code across your model and route files.

What if, instead of returning a dictionary, you create a variable that references it, you can create the dictionary, then check some logic and add `goal_id` if needed. 
```py
task_result = {
    "id" : self.id,
    "title": self.title,
    "description": self.description,
    "is_complete": self.completed_at is not None
}

if self.goal_id:
    task_result["goal_id"] = self.goal_id
    
return task_result
```